src/ts/components/subtitleoverlay.ts

```diff
@@ -59,34 +63,36 @@ export class SubtitleOverlay extends Container<ContainerConfig> {
       this.preprocessLabelEventCallback.dispatch(event, labelToAdd);
 
       if (this.previewSubtitleActive) {
-        this.removeComponent(this.previewSubtitle);
+        this.subtitleContainerManager.removeLabel(this.previewSubtitle);
       }
-      this.addComponent(labelToAdd);
-      this.updateComponents();
 
+      this.subtitleContainerManager.addLabel(labelToAdd, event);
```

Comment:
Passing the `event` here should be unnecessary... the `labelToAdd = subtitleManager.cueEnter(event);` call above should attach the information directly to the `SubtitleLabel`,